### PR TITLE
PR template update

### PR DESCRIPTION
**Pull request by JoeOster**
## Description

Updating the PR Template to reflect new requirements

### CODE REVIEW

**Comment by JoeOster** [comment]
Hey All, I am asking for reviews to see if anyone wants to add or modify this further. Swe had three new requirements due to SDLe and possibly a few more in a few weeks. Once I get this template finalized I will also be updating the wiki to reflect the new requirements and other minor changes

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 1]
```diff
@@ -24,21 +31,40 @@ Please describe the tests that you ran to verify your changes. Provide instructi
...
+- [ ] When compiling the compliler flag was used ""enable /Wall -Wformat-security -Werror=format-security"
+
+**_Delete this line and everything below if this is not a PR for a new sample_**
+
+**_Delete this line and all above it if this PR is for a new Sample_**
+# Adding a New samples
+##Introduction
 
-## Checklist for New samples:
+The full instructions on submitting a new sample is located in the [wiki](https://github.com/oneapi-src/oneAPI-samples/wiki/Contributing-a-New-Sample) if you have any questions about the steps for submitting a new code sample
```
Comment (mdbtucker) [inline .github/PULL_REQUEST_TEMPLATE.md:42]: This seems like an incomplete sentence.  If you have any questions about the steps for submitting a new code sample then . . . what?  If you're just trying to say read the wiki if you have questions, then I would just stop at 'The full instruction on submitting a new sample are (not is) located in the [wiki](link).'

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 2]
```diff
@@ -24,21 +31,40 @@ Please describe the tests that you ran to verify your changes. Provide instructi
...
 
+## Checklist
+Administrative
 - [ ] Review sample design with domain reviewers https://github.com/oneapi-src/oneAPI-samples/wiki/Reviewers-and-Domain-Experts
```
Comment (mdbtucker) [inline .github/PULL_REQUEST_TEMPLATE.md:46]: I've always found this one to be odd.  Generally people creating PRs don't have the power to select/add reviewers.  I would suggest leaving this out?

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 3]
```diff
@@ -24,21 +31,40 @@ Please describe the tests that you ran to verify your changes. Provide instructi
...
+##Introduction
 
-## Checklist for New samples:
+The full instructions on submitting a new sample is located in the [wiki](https://github.com/oneapi-src/oneAPI-samples/wiki/Contributing-a-New-Sample) if you have any questions about the steps for submitting a new code sample
 
+## Checklist
+Administrative
 - [ ] Review sample design with domain reviewers https://github.com/oneapi-src/oneAPI-samples/wiki/Reviewers-and-Domain-Experts 
+- [ ] Review Sample with Domain Expert: <insert NameHere>
+- [ ] If you dependencies/binaries, inform Samples Project Manager Swapna R Dontharaju (@srdontha) or @JoeOster
```
Comment (mdbtucker) [inline .github/PULL_REQUEST_TEMPLATE.md:48]: or Joe Oster (@JoeOster) - for consistency

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 4]
```diff
@@ -24,21 +31,40 @@ Please describe the tests that you ran to verify your changes. Provide instructi
...
+
+Code Development
 - [ ] Implement coding guidelines and ensure code quality.
```
Comment (mdbtucker) [inline .github/PULL_REQUEST_TEMPLATE.md:51]: this should have a link to the coding guidelines

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 5]
```diff
@@ -24,21 +31,40 @@ Please describe the tests that you ran to verify your changes. Provide instructi
...
-- [ ] Adhere to sample.json specification. https://github.com/oneapi-src/oneAPI-samples/wiki/sample-json-specification
-- [ ] Run jsonlint on sample.json to verify json syntax. www.jsonlint.com
 - [ ] Adhere to readme template 
-- [ ] Ensure/create CI test configurations for sample (ciTests field) https://github.com/oneapi-src/oneAPI-samples/wiki/sample-json-ci-test-object
 - [ ] Enforce format via clang-format config file
+- [ ] Adhere to sample.json specification. https://github.com/oneapi-src/oneAPI-samples/wiki/sample-json-specification
+- [ ] Ensure/create CI test configurations for sample (ciTests field) https://github.com/oneapi-src/oneAPI-samples/wiki/sample-json-ci-test-object
+- [ ] Run jsonlint on sample.json to verify json syntax. www.jsonlint.com
```
Comment (mdbtucker) [inline .github/PULL_REQUEST_TEMPLATE.md:56]: Is this not part of the CI?  If so, I don't think we need this in the checklist.

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 6]
```diff
@@ -24,21 +31,40 @@ Please describe the tests that you ran to verify your changes. Provide instructi
...
+
+Security and Legal
+- [ ] OSPDT Approval (see @JoeOster for assistance)
 - [ ] Compile code using compiler flags and fix anything detected "enable /Wall -Wformat-security -Werror=format-security"
```
Comment (mdbtucker) [inline .github/PULL_REQUEST_TEMPLATE.md:60]: Shouldn't this be "-Wall -Wformat-security -Werror=format-security"?

Also I'd change the wording slightly to 'Compile using the following compiler flags and fix any warnings'

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 1 above)
Comment (JoeOster) [reply .github/PULL_REQUEST_TEMPLATE.md]: Fixed

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 2 above)
Comment (JoeOster) [reply .github/PULL_REQUEST_TEMPLATE.md:46]: Not really adding it, so much as inserting a name. I think the wiki article says that and I will remove it

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 3 above)
Comment (JoeOster) [reply .github/PULL_REQUEST_TEMPLATE.md:48]: cleaned this up

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 4 above)
Comment (JoeOster) [reply .github/PULL_REQUEST_TEMPLATE.md:51]: good point-fixed

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 5 above)
Comment (JoeOster) [reply .github/PULL_REQUEST_TEMPLATE.md:56]: yes and no... 
The first  two are just referring to proper json construction
1. Sample browser setup
2. actually running the ci tests
3. the jsonint is preventative to keep the ci from failing

**Comment by mdbtucker** [comment]
@JoeOster did you forget about this PR, or is there something blocking you?

**Comment by JoeOster** [comment]
I need you to approve, so I can merge or you can

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 7]
```diff
@@ -24,21 +31,39 @@ Please describe the tests that you ran to verify your changes. Provide instructi
...
 - [ ] Enforce format via clang-format config file
-- [ ] Compile code using compiler flags and fix anything detected "enable /Wall -Wformat-security -Werror=format-security"
-- [ ] Review Sample with Domain Expert: <insert NameHere>
+- [ ] Adhere to sample.json specification. https://github.com/oneapi-src/oneAPI-samples/wiki/sample-json-specification
+- [ ] Ensure/create CI test configurations for sample (ciTests field) https://github.com/oneapi-src/oneAPI-samples/wiki/sample-json-ci-test-object
+- [ ] Run jsonlint on sample.json to verify json syntax. www.jsonlint.com
+
+Security and Legal
+- [ ] OSPDT Approval (see @JoeOster for assistance)
+- [ ] Compile using the following compiler flags and fix any warnings, the falgs are: "/Wall -Wformat-security -Werror=format-security"
```
Comment (mdbtucker) [inline .github/PULL_REQUEST_TEMPLATE.md:59]: I still think this should be -Wall not /Wall (especially in light of the new dpcpp/dpcpp-cl changes)

**Review by mdbtucker — DISMISSED** [review]

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 7 above)
Comment (JoeOster) [reply .github/PULL_REQUEST_TEMPLATE.md:59]: My apologies, I thought I had fixed it

**Review by mdbtucker — APPROVED** [review]

**Pull request merged**